dc/s2n-quic-dc/src/path/secret/map.rs

```diff
@@ -1018,12 +1020,21 @@ impl dc::Path for HandshakingPath {
                 .expect("peer tokens are only received after secrets are ready"),
             sender,
             receiver,
-            self.parameters,
+            self.parameters.clone(),
             self.map.state.rehandshake_period,
         );
         let entry = Arc::new(entry);
         self.map.insert(entry);
     }
+
+    fn on_mtu_updated(&mut self, mtu: u16) {
+        if let Some(entry) = self.map.state.peers.get_by_key(&self.peer) {
```

Comment:
I wonder if we should store an `Option<Arc<Entry>>` on `self` for this... The way it is right now may have issues if we have some other handshake that are racing each other.

Reply:
I think we'd probably need that for https://github.com/aws/s2n-quic/issues/2314 too?

Reply:
possibly, though i haven't looked into that issue too much at this point